cloudfoundry/cfapi/service_manager.go

```diff
@@ -505,6 +507,97 @@ func (sm *ServiceManager) ReadServiceInstance(serviceInstanceID string) (service
 	return serviceInstance, nil
 }
 
+// WaitServiceInstanceTo -
+func (sm *ServiceManager) WaitServiceInstanceTo(operationType string, serviceInstanceID string) (err error) {
+	sm.log.UI.Say("Waiting for service instance %s to finish starting ..", terminal.EntityNameColor(serviceInstanceID))
+
+	c := make(chan error)
+	go func() {
+
+		var err error
+		var serviceInstance CCServiceInstance
+
+		for {
+			time.Sleep(appStatePingSleep)
+			if serviceInstance, err = sm.ReadServiceInstance(serviceInstanceID); err != nil {
+				c <- err
+				return
+			}
+
+			if serviceInstance.LastOperation["type"] == operationType {
+				state := serviceInstance.LastOperation["state"]
+
+				switch state {
+				case "succeeded":
+					c <- nil
+					return
+				case "failed":
+					c <- fmt.Errorf("service instance %s crashed", serviceInstanceID)
+					return
+				}
+			}
+		}
+	}()
+
+	select {
+	case err = <-c:
+		if err != nil {
+			return err
+		}
+		sm.log.UI.Say("%s finished starting ...", terminal.EntityNameColor(serviceInstanceID))
+	}
+	return nil
+}
+
+// WaitDeletionServiceInstance -
+func (sm *ServiceManager) WaitDeletionServiceInstance(serviceInstanceID string) (err error) {
+	sm.log.UI.Say("Waiting for service instance %s to delete ..", terminal.EntityNameColor(serviceInstanceID))
+	c := make(chan error)
+	go func() {
+
+		var err error
+		var serviceInstance CCServiceInstance
+
+		for {
+			time.Sleep(appStatePingSleep)
+			if serviceInstance, err = sm.ReadServiceInstance(serviceInstanceID); err != nil {
+				// if the service instance is gone the error message should contain 60004
+				// cf_service_instance.redis: Server error, status code: 404, error code: 60004, message: The service instance could not be found: babababa-d977-4e9c-9bd0-4903d146d822
+				if strings.Contains(err.Error(), "error code: 60004") {
```

Comment:
Minor alternative wording suggestion

// if the service instance is gone the error message should contain error code 60004 ("ServiceInstanceNotFound")
+				// e.g. CLI output: cf_service_instance.redis: Server error, status code: 404, error code: 60004, message: The service instance could not be found: babababa-d977-4e9c-9bd0-4903d146d822